src/compiler/nvcc.rs

```diff
@@ -117,9 +117,13 @@ impl CCompilerImpl for NVCC {
 
         //NVCC only supports `-E` when it comes after preprocessor
         //and common flags.
-        cmd.arg("-E")
-            .arg("-Xcompiler=-P")
-            .env_clear()
+        cmd.arg("-E");
+        if cfg!(windows) {
+            cmd.arg("-Xcompiler=-EP");
```

Comment:
Could you please add a comment to explain what you are doing?

Reply:
If windows, `-Xcompiler=-P` will send precompiled content to a file, but hash key will generatate from stdout, so that hash key is wrong and remain the same. `-Xcompiler=-E` or  `-Xcompiler=-EP` will send precompiled content to stdout, it's right.

Reply:
Sorry, i meant in the code :)

Reply:
Hey, thanks for this patch, your explanation in #1017 makes a lot of sense, good investigation :+1: 
I have one question:

> To suppress the generation of #line directives, use [/EP (Preprocess to stdout Without #line Directives)](https://docs.microsoft.com/en-gb/cpp/build/reference/ep-preprocess-to-stdout-without-hash-line-directives?view=msvc-160) as well as /P.

Should we be providing both `-EP` and `-P`? Otherwise, will we get `stdout`, but `file.cpp.i` won't be created, right? Unsure if I'm understanding this correctly.

I agree with Sylvestre though: let's add a little comment in the code to explain what's going on:
```suggestion
            // The hash key is generated from stdout, so do "-EP" to print the output.
            cmd.arg("-Xcompiler=-EP");
```